src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -4984,6 +4985,38 @@ private SourceLocation getSelectionAsSourceLocation(boolean fromClick)
       int column = selPos.getColumn() + 1;
       return SourceLocation.create(file, line, column, fromClick);
    }
+   
+   @Handler
+   void onQuickAddNext()
+   {
+      if (!(docDisplay_ instanceof AceEditor))
```

Comment:
Even if you just hid most of the implementation within the AceEditor that would do it (i.e. just move the code over the boundary into DocDisplay). We've never taken advantage of it, but in principle the reason for representing AceEditor as DocDisplay is so that we can inject a test mock. Again, we've never used this so DocDisplay might as well just be AceEditor but it seems like we should avoid the cast if we can (again, not refactoring or exposing more internals, just move the code over the line).

If I'm not making sense then I probably just don't have a good enough handle on the use case and you can feel free to ignore!